new_version.sh

```diff
@@ -1,24 +1,48 @@
 #!/bin/bash
 
+has_breaking_exclamation() {
+  commit_message="$1"
+  commit_part="${commit_message%%:*}"
+
+  [[ $commit_part == *! ]]
+}
+
+has_breaking_footer() {
+  commit_message="$1"
+  breaking_token1="BREAKING CHANGE"
+  breaking_token2="BREAKING-CHANGE"
+
+  if [[ "$commit_message" == *$'\n\n'"$breaking_token1: "* ]]
+  then true
+  elif [[ "$commit_message" == *$'\n\n'"$breaking_token2: "* ]]
+  then true
+  else false
+  fi
+}
+
+is_breaking_change() {
+  has_breaking_footer "$1" || has_breaking_exclamation "$1"
+}
+
 commit_message="$1"
 current_version="$2"
 echo "Obtaining new version from commit message \"$commit_message\" and current version \"$current_version\""
 
 current_version_major=$(echo $current_version | cut -d. -f1)
 current_version_minor=$(echo $current_version | cut -d. -f2)
-current_version_revision=$(echo $current_version | cut -d. -f3)
+current_version_patch=$(echo $current_version | cut -d. -f3)
 
-commit_part=$(echo $commit_message | cut -d "(" -f1)
+commit_type=$(echo "${commit_message%%:*}" | cut -d "(" -f1)
```

Comment:
This does not comply with the supported format of `<type>[optional scope]: <description>`.
This probably does the trick, as shown below.
```suggestion
commit_type=$(echo "${commit_message%%:*}" | cut -d ":" -f1 | cut -d "(" -f1)
```

```
❯ echo "feat: My feature" | cut -d "(" -f1
feat: My feature

~
❯ echo "feat(#1): My feature" | cut -d "(" -f1
feat

~
❯ echo "feat(#1): My feature" | cut -d ":" -f1 | cut -d "(" -f1
feat

~
❯ echo "feat: My feature" | cut -d ":" -f1 | cut -d "(" -f1 
feat

~
❯ echo "feat: My feature" | cut -d ":" -f1 | cut -d "(" -f1
feat
```

Reply:
This is not needed, as the first part gets that substring (before `:`):
```
${commit_message%%:*}
```